### DISCUSSION TITLE
Angular web worker cannot import akita

### DISCUSSION THREAD

**Discussion by kragoth** [salesforce/akita#678] [Q&A]
A while back I was working on moving our Akita persistence into a webworker. Unfortunately due to time constraints I had to stop working on this, until now.
Previously I had a very primitive example of Akita working in a webworker.

Unfortunately, it appears that it is no longer possible to use Akita in an Angular webworker.
I've created a [StackBlitz](https://stackblitz.com/edit/angular-uphsdg?) to reproduce the problem. 
WebWorkers don't work in stackblitz so, the project will need to be downloaded and then do the usual
```
npm i
npm run ng build
```

This will fail with the following error:
```
Error: ./src/app/web-workers/test.worker.ts (./node_modules/worker-plugin/dist/loader.js?name=0!./src/app/web-workers/test.worker.ts)
Module build failed (from ./node_modules/worker-plugin/dist/loader.js):
Error: node_modules/@angular/core/core.d.ts:10411:62 - error TS2304: Cannot find name 'Element'.

10411 export declare function ɵgetDebugNode__POST_R3__(nativeNode: Element): DebugElement__POST_R3__;
```

If you then open src/app/web-workers/test.worker.ts and comment out the import of the akita libraries the error goes away.

So, somehow when Akita is imported it seems to cause the @angular/core library to also be imported?
Or am I doing something totally wrong?

I was excited to finally get back to trying this today, only to be hit with this problem which I have no idea how to even debug.

**Comment by kragoth** [comment]
OK, I think I've made a bit of progress on where the problem lies.
At the end of the day I don't think this is an Akita problem.

The problem is with Angular. When you run an ng build the store.d.ts file is modified by angular for some reason.
Just above the class declaration become:
```
import * as ɵngcc0 from '@angular/core';
export declare class Store<S = any> {
```

The lasts few lines of the class become:
```
    static ɵfac: ɵngcc0.ɵɵFactoryDef<Store<any>, never>;
    static ɵdir: ɵngcc0.ɵɵDirectiveDefWithMeta<Store<any>, never, never, {}, {}, never>;
}
```

So.... I guess I'm off to file a bug with Angular.

**Reply by kragoth** [reply]
https://github.com/angular/angular/issues/42139 is the angular issue if anyone else runs into this problem.

**Reply by kragoth** [reply]
@NetanelBasal you might find the discussion of this issue over on the angular github interesting. Basically this is occurring because store.d.ts has the ngOnDestroy function defined.

**Reply by NetanelBasal** [reply]
Actually, this is a legacy code left from Akita's beginning when used only with Angular applications. I think we'll remove it on the next major version.
